### PR TITLE
ub configs (for performance)

### PR DESCRIPTION
**Pull request by malay-nagda**
# What does this PR do ?

tb overlap configs for llama3.1 405b, gpt 20b and 175b

**Collection**: [llm]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
  trainer = run.Config(
        nl.Trainer,
        callbacks = [
            run.Config(
                    MegatronCommOverlapCallback,
                    tp_comm_overlap=True,
                    tp_comm_overlap_cfg=userbuffers_fp8_h100_h12288_tp4_mbs1_seqlen2048,
             )
        ]
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/recipes/tp_overlap_configs/userbuffers.py [hunk 1]
```diff
@@ -20,6 +20,8 @@ class PipelineOverlapCfg(TPOverlapCfg):
 class RingExchangeOverlapCfg(TPOverlapCfg):
     aggregate: bool = False
     method: str = 'ring_exchange'
+    num_sm: int = 1
+    set_sm_margin: bool = False
```
Comment (malay-nagda) [inline nemo/collections/llm/recipes/tp_overlap_configs/userbuffers.py:24]: Added this according to config in [NeMo-Framework-Launcher](https://github.com/NVIDIA/NeMo-Framework-Launcher/blob/main/launcher_scripts/conf/training/tp_overlap/ub_cfg_h100_h12288_tp4_mbs1_seqlen2048.yaml#L57)

Comment (malay-nagda) [reply nemo/collections/llm/recipes/tp_overlap_configs/userbuffers.py:24]: Added defaults according to [T.E base.by](https://github.com/NVIDIA/TransformerEngine/blob/main/transformer_engine/pytorch/module/base.py#L248)

**Review by JimmyZhang12 — APPROVED** [review]
lgtm!

**Review by erhoo82 — APPROVED** [review]
LGTM

**Pull request merged**